tests/unit/changeset-test.js

```diff
@@ -1780,15 +1780,32 @@ module('Unit | Utility | changeset', function (hooks) {
   test('observing #rollback values', async function (assert) {
     let res;
     let changeset = Changeset(dummyModel, dummyValidator);
-    changeset.addObserver('name', function () {
-      res = this.name;
+    //To consistently observe for changes (even for deep keys),
+    //we must adhere to a never changing value
+    //so the changeset.data (_content) is the silver bullet for now
+    //ember-changeset will keep notifying on it
+    //one downside is that you won't get the changeset as `this` inside the callback.
+    changeset.data.addObserver('name', function () {
+      res = changeset.get('name');
     });
     assert.equal(undefined, changeset.get('name'), 'initial value');
     changeset.set('name', 'Jack');
     assert.equal('Jack', res, 'observer fired when setting value');
     changeset.rollback();
     assert.equal(undefined, res, 'observer fired with the value name was rollback to');
   });
+  test('observing deep keys on .data works', async function (assert) {
+    let res;
+    let changeset = Changeset(dummyModel, dummyValidator);
+    changeset.data.addObserver('some.really.nested.path', function () {
+      res = changeset.get('some.really.nested.path');
+    });
+    assert.equal(undefined, changeset.get('some.really.nested.path'), 'initial value');
+    changeset.set('some.really.nested.path', 'Jack');
+    assert.equal('Jack', res, 'observer fired when setting value some.really.nested.path');
+    changeset.rollback();
+    assert.equal(undefined, res, 'observer fired with the value some.really.nested.path was rollback to');
+  });
 
   test('can update nested keys after rollback changes.', async function (assert) {
     let expectedResult = {
```

Comment:
https://github.com/poteto/ember-changeset/issues/585

Do we have a test for this issue so we can close it out when merged?

Reply:
ping!  Lmk what you think about adding this test.  Lots of great work in this PR.

Reply:
sorry for the delay, I ran that repo, I think it has a bug on the example itself, it uses 

```hbs
@onToggle={{action (changeset-set this.changesetObj contextKey) (not (changeset-get this.changesetObj contextKey))}}
```

I think that action helper is actually(?) mutating the CONTENT, somehow(?) and thus _it actually have changes_

but I fix I just ran this branch with a few changes to the example and its working!

```hbs
@onToggle={{changeset-set
          this.changesetObj
          contextKey
          (not (changeset-get this.changesetObj contextKey))
        }}


```

Reply:
And so, I think... no further test need to be added? 🤔  I mean there are a few tests already checking for changes emptied after setting back the initial wrapped value? gonan double check

Reply:
Hmm I ran this again and the bug still exists, but I'm unable to track _why_ it seems somehow `sets` are applied to the this[CONTENT] without executing the changeset using `changeset-set`.